### DISCUSSION TITLE
How to preserve scrollposition of page after returning from another?

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#18819] [Questions] [answered]
What I have tried so far (after searching a lot through docs and discussions):

```js
// plugins/scrollbehaviour.client.js
export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.$router.options.scrollBehavior = (to, from, savedPosition) => {
        console.log(to.name, savedPosition);

        if (to.name === "index" && savedPosition) {
            return savedPosition;
        }

        return { top: 0 };
    };
});
```

The problem (I guess) is that `savedPosition` always returns null.

Why is that? And how can I simply preserve the last scroll position for a page after returning from another?

**Comment by TheAlexLichter** [comment]
You would change it via `app/router.options.ts`, see e.g. the [default one](https://github.com/nuxt/framework/blob/main/packages/nuxt/src/pages/runtime/router.options.ts).

**Reply by ghost** [reply]
Yes, I can also do it via `router.options.ts` but the question is how to achieve it. I have figured out that according to the docs …
> savedPosition is only available for popstate navigations (back button)

So I need to figure out if the user comes from this page …

**Comment by ghost** [comment, marked as answer]
You will find a solution here: https://github.com/nuxt/nuxt/discussions/18846
